doc/metrics/maximum-nesting.md

```diff
@@ -0,0 +1,33 @@
+# Maximum Nesting
+
+**Maximum Nesting** is the maximum nesting level of blocks / control structures that are present in a method. Methods with deep nesting level are often complex and tough to maintain.
```

Comment:
@dkrutskikh documentation mentions only methods and tests operate with only function. I'd suggest to add functions example to docs and methods example to tests then.

Reply:
@incendial texts updated